### ISSUE TITLE
unable to install Lombok

### ISSUE THREAD

**Issue by alculete** [eclipse-linuxtools/eclipse-ide-snap#2]
Unable to install Lombok or any other plugin not in the store through **Help > Install New Software**.
It finished installing, but after that, it restarted, and nothing was installed. This might be related to **eclipse.ini** not being writable.

**Comment by Mailaender** [comment]
What is added to the `eclipse.ini` on a tarball installation?

**Comment by alculete** [comment]
it adds a javaagent at the end of the file. if there is no write permission you can´t change the file maybe eclipse.ini could be put on a different location, maybe on ~/snap/eclipse. This is done when you install the addon. In this case the Lombok addon, but I think it might work the same way as other addons installed through Install new Software.

![image](https://github.com/user-attachments/assets/e12e4881-91c3-4d07-95cd-7f54252a16a1)

**Comment by Mailaender** [comment]
Does it work when you manually modify according to:
https://github.com/eclipse-linuxtools/eclipse-ide-snap/blob/ab8bc96ca5816a61a9ccef21982716a4f5824050/snap/local/wrappers/eclipse#L7

**Comment by alculete** [comment]
Yes, it does. Can´t eclipse.ini be changed from    `   'eclipse.ini': 'usr/lib/eclipse/eclipse.ini'` to       '`eclipse.ini': 'home/user/snap/eclipse/eclipse.ini'` where it could be written. Sorry if this doesn't make sense or it is impossible to be done, I have never packed a snap before

**Comment by Mailaender** [comment]
Try changing `$ECLIPSE_HOME/eclipse.ini` as suggested in https://wiki.eclipse.org/Eclipse.ini See https://github.com/eclipse-linuxtools/eclipse-ide-snap#development on how to build snaps locally. You might need to `snapcraft clean` to force a fresh file system for the package build.

**Comment by alculete** [comment]
ok Will try and revert. Also, a reminder that eclipse 2024-12 nog running

**Comment by Mailaender** [comment]
https://github.com/eclipse-linuxtools/eclipse-ide-snap/issues/5 seems to be a you problem. Maybe because you are manually controlling revisions that disabled the automatic update.

**Comment by alculete** [comment]
I downgraded it because it updated automatically and stopped working. Purging solved the issue

**Comment by alculete** [comment]
Hi @Mailaender  I know what caused the issue. When the IDE is updated the eclipse.ini on /home/user/snap/... is not updated and the plugins listed there became outdated comparing to the one int he original eclipse.ini. One disadvantage of overriding the default eclipse.ini.

**Comment by Mailaender** [comment]
It is definitely not an optimal solution to the read-only file system problem that was inherited from @snapcrafters as stated in https://github.com/snapcrafters/eclipse/issues/74 and  https://github.com/projectlombok/lombok/issues/2720 but I don't have a better solution at hand.

**Comment by alculete** [comment]
The first link you provided was an issue I reported. I again with 2024-12 it is not working.
Unfortunately, I will have to refactor and manually add the boilerplate code to my code or move the compressed file

**Comment by alculete** [comment]
Any progress in this?

**Comment by Mailaender** [comment]
No. Maybe this is even hitting a fundamental limitation of sandboxing and atomic updates. I created https://mpr.makedeb.org/packages/eclipse-rcp-bin as an alternative without snap.

**Comment by alculete** [comment]
I think one of the issues is that Eclipse has some plugins outside the marketplace.
We have other IDEs that package the same way, working well

**Comment by alculete** [comment]
Hi @Mailaender  is this close as not solved?

**Comment by Mailaender** [comment]
Now it should be impossible as to my knowledge you can't edit the `eclipse.ini` in the readonly file system.

**Comment by alculete** [comment]
Thanks for the feedback
